app/components/ContextMenu.tsx

```diff
@@ -176,6 +177,37 @@ const OptionViewArtist = React.memo<{
   )
 })
 
+const OptionShuffleArtist = React.memo<{
+  artistId?: string
+}>(({ artistId }) => {
+  if (!artistId) {
+    return <></>
+  }
+
+  // TODO This hook gets called whenever the context menu is opened.
+  // That is wasteful as it fetches a bunch of information which might not
+  // be used. We should only fire this hook if the user actually clicks shuffle
+  // but that breaks Rules Of Hooks and I'm a react noob.
+  const artist = useArtistWithSongs(artistId)
```

Comment:
Instead of having the artist and songs fetched in a hook when the component loads here, can we instead create a hook that returns an async function that fetches that info and then calls `setQueue` so we can assign it to the button?

That way this option can exist in the context menu regardless of data needing to be fetched as it pops up as well, which avoids the awkward adding of an option to the menu/changing its size after it has already been shown.

Reply:
I understand this conceptually...implementing it properly in React is another thing.

My last commit tries to do this and I think it's working properly, but I may be horrendously misusing the `useStore` hook.